protocols/gossipsub/Cargo.toml

```diff
@@ -10,6 +10,9 @@ repository = "https://github.com/libp2p/rust-libp2p"
 keywords = ["peer-to-peer", "libp2p", "networking"]
 categories = ["network-programming", "asynchronous"]
 
+[features]
+wasm-bindgen = ["getrandom/js", "instant/wasm-bindgen"]
```

Comment:
We have a `wasm-bindgen` feature in the `libp2p` meta crate. Can you activate this from there please?

Also, you can remove the `getrandom` dependency here because we already activate that one in the `libp2p` meta-crate!

Reply:
> We have a wasm-bindgen feature in the libp2p meta crate. Can you activate this from there please?

Done.

> Also, you can remove the getrandom dependency here because we already activate that one in the libp2p meta-crate!

I can't enable it's feature if I remove it.

Reply:
> > Also, you can remove the getrandom dependency here because we already activate that one in the libp2p meta-crate!
> 
> I can't enable it's feature if I remove it.

The meta-crate enables it for you! :)

https://github.com/libp2p/rust-libp2p/blob/master/libp2p/Cargo.toml#L83

Reply:
I think that will not work when we use gossipsub directly as a crate. Am I missing something?

Reply:
e.g,

```toml
libp2p-gossipsub = { git = "https://github.com/ozkanonur/rust-libp2p.git", branch = "t", features = ["wasm-bindgen"] }
```

Because of https://github.com/ozkanonur/rust-libp2p/commit/88b39dc3fab80004e524f65ab19bd782ac725a9d this change, it will not compile due to missing `js` feature of getrandom.

Reply:
If `libp2p-gossipsub` must be pulled with `libp2p` meta-crate together, then I can do that. I can remove `features` section in `libp2p-gossipsub`, in fact I can even remove `getrandom` and `instant` dependencies in `libp2p-gossipsub` because they already comes with the meta-crate. But if `libp2p` is missing in the dependency tree, then `gossipsub` will not be able to compile successfully. Both fine for me, I leave this up to you:)

Reply:
Perhaps the `wasm-bindgen` feature in `libp2p` is also wrong but the way I'd see it is that depending on `getrandom` and activating `js` is something that users can always do themselves. We don't need to forward that feature-flag for them.

I don't know to be honest :D

At some point we will have to sit down and work out, what the best way is to handle these features. Every crate seems to solve WASM-compatibility in a different way so it is unclear what the correct way is.

Reply:
> Perhaps the wasm-bindgen feature in libp2p is also wrong

Yes, I have noticed some of the issues as well. However, I didn't want to make extensive changes in this pull request. Perhaps I will address them in a separate one at a later time.